makeflow/src/makeflow_gc.c

```diff
@@ -83,6 +83,8 @@ void makeflow_parse_input_outputs( struct dag *d )
 		/* add collect_list, for sink_files that should be removed */
 		string_split_quotes(input_list, &argc, &argv);
 		for(i = 0; i < argc; i++) {
+			// skip empty strings
+			if(!argv[i][0]) continue;
```

Comment:
Aha, here is the difference.
In the old version, the empty string was considered to be a "file" and counted among the "completed files".
As a result, GC based on a *count* of files happened a little too early.